src/Microsoft.SqlTools.ServiceLayer/ObjectManagement/ObjectTypes/Database/DatabaseHandler.cs

```diff
@@ -401,8 +401,7 @@ private string ConfigureDatabase(InitializeViewRequestParams viewParams, Databas
                         prototype.Name = database.Name;
 
                         // Update database file names now that we have a database name
-                        // Modifying logical file name is not supported in SQL Database Managed Instance.
-                        if (!prototype.HideFileSettings && dataContainer.Server.DatabaseEngineEdition != DatabaseEngineEdition.SqlManagedInstance)
```

Comment:
Removing MI condition, as it was added to fix the same issue for MI db properties, but the IsNewObject check fixes all cases.